### ISSUE TITLE
Popup window does not resize any longer after scrolling

### ISSUE THREAD

**Issue by lifepillar** [vim/vim#14745]
### Steps to reproduce

0. Run `vim --clean`
1. Source this script and take note of the reported popup id (let's say it's `1001`):

```vim
vim9script

def Popup(): number
  return popup_create([], {
    border: [1, 1, 1, 1],
    close: 'click',
    minheight: 1,
    maxheight: 10,
    scrollbar: true,
    minwidth: &columns - 5,
    maxwidth: &columns - 5,
  })
enddef

var id = Popup()
echo id

popup_settext(id, repeat(['abcd'], 20))
```

2. With the mouse, scroll down the popup. This step is crucial to reproduce the issue!
3. Execute `:call popup_settext(1001, [])`. The popup becomes empty and with minimum height.
4. Execute `:call popup_settext(1001, ['a', 'b', 'c'])`

The height of the popup does not change at step 4.

### Expected behaviour

I would expect the popup to resize at step 4 to fit the new content. That is what happens if scrolling (step 2) is skipped when performing the above.

### Version of Vim

9.1.405

### Environment

macOS 14.4.1
Apple Terminal
xterm-256color
ZSH 5.9

**Comment by habamax** [comment]
If I scroll with mouse after step 4, window height increases.

**Comment by lifepillar** [comment]
That's right, scrolling down restores the correct height. But scrolling should not be necessary. If you repeat the steps above skipping step 2, at step 4 the height is restored at once, as expected. I would expect that any call to `popup_settext()` adapts the popup height (within the `minheight`/`maxheight` limits) regardless of the cursor's position.
